### PR TITLE
feat: add view fixed deposit product component

### PR DESCRIPTION
**Pull request by spider0061**
add view fixed deposit product component.

## Related issues and discussion
#465 

## Screenshots, if any
Screenshot from 2020-04-29 18-00-52

## Checklist
Please make sure these boxes are checked before submitting your pull request - thanks!

- [x] If you have multiple commits please combine them into one commit by squashing them.

- [x] Read and understood the contribution guidelines at `web-app/.github/CONTRIBUTING.md`.

### CODE REVIEW

File: src/app/products/fixed-deposit-products/fixed-deposit-product.resolver.ts [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+/** rxjs Imports */
+import { Observable } from 'rxjs';
+
+/** Custom Services */
+import { ProductsService } from '../products.service';
+
+/**
+ * Fixed Deposit Product data resolver.
+ */
+
```
Comment (OussEmaDevCode) [inline src/app/products/fixed-deposit-products/fixed-deposit-product.resolver.ts]: remove this line

File: src/app/products/fixed-deposit-products/view-fixed-deposit-product/view-fixed-deposit-product.component.ts [hunk 2]
```diff
@@ -0,0 +1,38 @@
+/** Angular Imports */
+import { Component, OnInit } from '@angular/core';
+import { ActivatedRoute } from '@angular/router';
+import { MatTable } from '@angular/material/table';
+import { MatDialog } from '@angular/material';
+
+/**
+ * Fixed Deposit Product component.
+ */
+
```
Comment (OussEmaDevCode) [inline src/app/products/fixed-deposit-products/view-fixed-deposit-product/view-fixed-deposit-product.component.ts]: remove this line

File: src/app/products/fixed-deposit-products/view-fixed-deposit-product/view-fixed-deposit-product.component.ts [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+@Component({
+  selector: 'mifosx-view-fixed-deposit-product',
+  templateUrl: './view-fixed-deposit-product.component.html',
+  styleUrls: ['./view-fixed-deposit-product.component.scss']
+})
+export class ViewFixedDepositProductComponent implements OnInit {
+
+  /** Fixed Deposit Product data. */
+
```
Comment (OussEmaDevCode) [inline src/app/products/fixed-deposit-products/view-fixed-deposit-product/view-fixed-deposit-product.component.ts]: remove this line

File: src/app/products/fixed-deposit-products/view-fixed-deposit-product/view-fixed-deposit-product.component.ts [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+  fixedDepositProductData: any;
+  hasAccounting: boolean;
+
+  interestPeriodDisplayedColumns: string[] = ['periodType', 'amountRange', 'interestRate', 'description', 'incentives'];
+  chargesDisplayedColumns: string[] = ['name', 'type', 'amount', 'collectedon'];
+  paymentFundSourceDisplayedColumns: string[] = ['paymentTypeId', 'fundSourceAccountId'];
+  feesPenaltyIncomeDisplayedColumns: string[] = ['chargeId', 'incomeAccountId'];
+
```
Comment (OussEmaDevCode) [inline src/app/products/fixed-deposit-products/view-fixed-deposit-product/view-fixed-deposit-product.component.ts]:
``` suggestion 

  /**
   * Retrieves the fixed deposit product data from `resolve`.
   * @param {ActivatedRoute} route Activated Route.
   */

File: src/app/products/products.service.ts [hunk 5]
```diff
@@ -135,6 +135,10 @@ export class ProductsService {
     return this.http.get('/fixeddepositproducts/template');
   }
```
Comment (OussEmaDevCode) [inline src/app/products/products.service.ts]:
``` suggestion
  /**
   * @param {string} fixedDepositProductId fixed deposit product ID of fixed deposit product.
   * @returns {Observable<any>} Fixed deposit product.
   */

**Comment by spider0061** [comment]
@OussEmaDevCode done

File: src/app/products/fixed-deposit-products/view-fixed-deposit-product/view-fixed-deposit-product.component.scss [hunk 6]
```diff
@@ -0,0 +1,24 @@
+table {
```
Comment (abhaychawla) [inline src/app/products/fixed-deposit-products/view-fixed-deposit-product/view-fixed-deposit-product.component.scss:1]: Please ensure proper indentation in the files.

File: src/app/products/fixed-deposit-products/view-fixed-deposit-product/view-fixed-deposit-product.component.html [hunk 7]
```diff
@@ -0,0 +1,421 @@
...
+        </div>
+
+        <div fxFlexFill *ngIf="fixedDepositProductData.withHoldTax">
+          <span fxFlex="40%">Withhold Tax group:</span>
+          <span fxFlex="60%">{{ fixedDepositProductData.taxGroup.name }}</span>
+        </div>
+
+        <hr>
+
+        <h3 class="mat-h3" fxFlexFill>Interest Rate Chart</h3>
```
Comment (abhaychawla) [inline src/app/products/fixed-deposit-products/view-fixed-deposit-product/view-fixed-deposit-product.component.html:156]: Have a look at the current implementation of Interest Rate Charts in the web app, in Preview of Create Fixed Deposit Product and update this accordingly. All interest rate chart slabs and incentives should be visible in table view here.

File: src/app/products/fixed-deposit-products/view-fixed-deposit-product/view-fixed-deposit-product.component.html [hunk 8]
```diff
@@ -0,0 +1,421 @@
...
+
+        <mat-divider [inset]="true"></mat-divider>
+
+        <div layout="row" layout-align="end center" flex>
+          <a href="#">View / Edit Interest Rate Charts</a>
+          <span flex></span>
+        </div>
```
Comment (abhaychawla) [inline src/app/products/fixed-deposit-products/view-fixed-deposit-product/view-fixed-deposit-product.component.html:163]: Remove this.

File: src/app/products/fixed-deposit-products/view-fixed-deposit-product/view-fixed-deposit-product.component.html [hunk 9]
```diff
@@ -0,0 +1,421 @@
...
+
+        <div fxFlexFill>
+          <span fxFlex="40%">Short Name:</span>
+          <span fxFlex="60%">{{ fixedDepositProductData.shortName }}</span>
+        </div>
+
+        <hr>
+
+        <h3 class="mat-h3" fxFlexFill>Items</h3>
```
Comment (abhaychawla) [inline src/app/products/fixed-deposit-products/view-fixed-deposit-product/view-fixed-deposit-product.component.html:39]: Make sure the fields are under appropriate headings.

File: src/app/products/fixed-deposit-products/view-fixed-deposit-product/view-fixed-deposit-product.component.html [hunk 10]
```diff
@@ -0,0 +1,421 @@
...
+
+        <div fxFlexFill>
+          <span fxFlex="40%">Default Deposit Amount:</span>
+          <span fxFlex="60%">{{ fixedDepositProductData.depositAmount | number}}</span>
+        </div>
+
+        <div fxFlexFill>
+          <span fxFlex="40%">Maximum Deposit Amount:</span>
+          <span fxFlex="60%">{{ fixedDepositProductData.maxDepositAmount | number}}</span>
+        </div>
```
Comment (abhaychawla) [inline src/app/products/fixed-deposit-products/view-fixed-deposit-product/view-fixed-deposit-product.component.html:71]: Take reference from the Preview of Create Fixed Deposit Product and update. Make sure all fields that are non-mandatory are only visible if their value exists.

File: src/app/products/fixed-deposit-products/view-fixed-deposit-product/view-fixed-deposit-product.component.html [hunk 11]
```diff
@@ -0,0 +1,421 @@
+<div fxLayout="row" fxLayoutAlign="end" fxLayoutGap="2%" fxLayout.lt-md="column" class="container m-b-20">
```
Comment (abhaychawla) [inline src/app/products/fixed-deposit-products/view-fixed-deposit-product/view-fixed-deposit-product.component.html:1]: See the components of Loan, Savings, Shares products as examples and update to maintain consistency wherever necessary. 

For example: for some cases where we have min, max, and default values they have been combined, Use of ngIf for conditional fields without creating any extra variable in the component etc.

**Comment by spider0061** [comment]
@abhaychawla Made the changes. Please have a look.

**Comment by karantakalkar** [comment]
@spider0061 resolve MCs please, I'll ask Abhay to complete review.

**Pull request merged**